### ISSUE TITLE
expected dense_16 to have shape (1,) but got array with shape (3,)

### ISSUE THREAD

**Issue by Kuselokusi** [keras-team/keras#12116]
Good-day, my dataset has 13 variables and 13109 rows. Test set has 5618 rows and same variable number. I have the following code:

```
import pandas as pd
import requests
import os
import io

from sklearn import metrics
from sklearn.model_selection import train_test_split #can be used to split data quickly into test and train sets, not used here
from keras.models import Sequential
from keras.layers.recurrent import SimpleRNN
from keras.layers.core import Dense, Activation, Dropout, Flatten
from keras.callbacks import EarlyStopping

model = Sequential()
#input layer
model.add(SimpleRNN(units=10, return_sequences=True, input_shape=(3, 13), activation ='sigmoid')) #input shape parameters 1 is for timesteps and 13 for number of indicators
model.add(Dropout(0.3))
#hidden layer 1
model.add(SimpleRNN(units=10, return_sequences=True, activation ='sigmoid'))  
model.add(Dropout(0.3))
#output layer for prediction
model.add(Flatten())
model.add(Dense(units=1, activation = 'sigmoid'))

model.compile(optimizer = 'adam', loss = 'binary_crossentropy') #use categorical crossentropy if more than one category predicted
#train model
model.fit(x_train_seq, y_train_seq, validation_split = 0.05, epochs = 100, verbose = 0, batch_size =32)
```
I have also created the required sequences of measurements using:

```
n = x_train.shape[0]
L = 3
x_train_seq = []
y_train_seq = []
for k in range(n - L + 1):
    x_train_seq.append(x_train[k : k + L])
    y_train_seq.append(y_train[k : k + L])

x_train_seq = np.array(x_train_seq)
y_train_seq = np.array(y_train_seq)
```
Kindly assist please, I don't get why i'm having this issue. y_train_seq is a categorical numpy array with on one variable to be predicted.

**Comment by tRosenflanz** [comment]
The issue is with how you prepare `y_train` (`y_train_seq.append(y_train[k : k + L])`) which will have shape (3,) if you do `y_train.shape`. 
Feel free to create a stackoverflow question - this is not a Keras issue
